options/cf_options.cc

```diff
@@ -297,10 +375,69 @@ std::unordered_map<std::string, OptionTypeInfo>
          {offset_of(&ColumnFamilyOptions::memtable_factory),
           OptionType::kMemTableRepFactory, OptionVerificationType::kByName,
           OptionTypeFlags::kNone, 0}},
+        {"memtable",
+         {offset_of(&ColumnFamilyOptions::memtable_factory),
+          OptionType::kMemTableRepFactory, OptionVerificationType::kAlias,
+          OptionTypeFlags::kNone, 0,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, char* addr) {
+            std::unique_ptr<MemTableRepFactory> rep;
+            Status s = GetMemTableRepFactoryFromString(value, &rep);
+            if (s.ok()) {
+              auto mtf =
+                  reinterpret_cast<std::shared_ptr<MemTableRepFactory>*>(addr);
+              mtf->reset(rep.release());
+            }
+            return s;
+          }}},
         {"table_factory",
          {offset_of(&ColumnFamilyOptions::table_factory),
           OptionType::kTableFactory, OptionVerificationType::kByName,
-          OptionTypeFlags::kNone, 0}},
+          OptionTypeFlags::kCompareLoose, 0}},
+        {"block_based_table_factory",
+         {offset_of(&ColumnFamilyOptions::table_factory),
+          OptionType::kTableFactory, OptionVerificationType::kAlias,
+          OptionTypeFlags::kCompareLoose, 0,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, char* addr) {
+            // Nested options
+            auto tf = reinterpret_cast<std::shared_ptr<TableFactory>*>(addr);
+            BlockBasedTableOptions table_opts, base_opts;
+            BlockBasedTableFactory* bbtf =
+                static_cast_with_check<BlockBasedTableFactory, TableFactory>(
+                    tf->get());
+            if (bbtf != nullptr) {
+              base_opts = bbtf->table_options();
+            }
+            Status s = GetBlockBasedTableOptionsFromString(base_opts, value,
+                                                           &table_opts);
+            if (s.ok()) {
+              tf->reset(NewBlockBasedTableFactory(table_opts));
+            }
+            return s;
+          }}},
+        {"plain_table_factory",
+         {offset_of(&ColumnFamilyOptions::table_factory),
+          OptionType::kTableFactory, OptionVerificationType::kAlias,
+          OptionTypeFlags::kCompareLoose, 0,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, char* addr) {
+            // Nested options
+            auto tf = reinterpret_cast<std::shared_ptr<TableFactory>*>(addr);
```

Comment:
In the Lambda expression, can you keep the original variable names such as plain_table_factory, table_opt? ptf and tf are hard to follow. Also for other cases like block_based_table_factory, memtable.

Reply:
> I am not sure what you mean by "original variable name"? Do you mean the name in the options structure or something else?

I mean, just replace tf, ptf with table_factory, plain_table_factory to make it more readable